src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -293,6 +294,84 @@ class AccessControlAttribute : public chip::app::AttributeAccessInterface
 
 constexpr uint16_t AccessControlAttribute::ClusterRevision;
 
+CHIP_ERROR LogAccessControlEvent(const AccessControl::Entry & entry, const Access::SubjectDescriptor & subjectDescriptor,
```

Comment:
What worries me with this approach is that the ACL store can change without any events being generated, if something other than this cluster changes it...  ideally event generation would happen via the ACL bits calling some callback when they change and the callback doing the "send the event" bits.

Reply:
There is only one case where ACL is created outside the cluster, and there will eventually be the case of removing a fabric or factory data reset, but we don't have that yet.

After that part if done, we will add the related callback here to do the "send the event" bits. Marc already has the idea how to do this.

Reply:
My question is what happens when the app uses SDK APIs to add/remove ACLs independently of the cluster.

Or will we not be exposing APIs for that?  If we're not going to expose general ACL mutation APIs, I am less worried about the event situation.  But my impression is that anyone can use the ACL module API, not just this cluster or AddNOC...

Reply:
We talked about that in DM and I believe we said we didn't want code to play with ACLs. This is why I've always been saying, the actual code that uses Access Control APIs is fairly limited.

I can't remember if we put that in the spec. AddNOC is an exceptional case, as stated in the spec; it's the one place an ACL is created, that isn't an administrator using the cluster.

As it stands, if you use the system APIs to play with APIs, it will work fine, without making events. If you use the cluster APIs, it will work fine, and make events.

Reply:
Right, my concern is that I'd really like to avoid ACLs changing without events, because that's a good way to sneak things by administrators...  My _preference_, obviously, would be for it to not be generally considered OK (and not made particularly easy) to mess with ACLs except via the approved clusters, which would auto-generate events.

Reply:
@bzbarsky-apple I captured this concern in detail in #14326 so if this PR is otherwise acceptable, is that OK?

Reply:
Yes, that works, thank you.